src/wp-includes/rest-api.php

```diff
@@ -1643,11 +1665,18 @@ function rest_validate_value_from_schema( $value, $args, $param = '' ) {
 		}
 
 		foreach ( $value as $property => $v ) {
+			$pattern_property_schema = rest_get_matching_pattern_property_schema( $property, $args );
```

Comment:
Can we not calculate this until after we know this is necessary? In other words, I'd like to avoid the regex checks if the property is specifically named in the `properties` list.